core/src/main/scala/org/apache/spark/internal/config/package.scala

```diff
@@ -1927,4 +1927,31 @@ package object config {
       .version("3.0.1")
       .booleanConf
       .createWithDefault(false)
+
+  private[spark] val PUSH_BASED_SHUFFLE_ENABLED =
+    ConfigBuilder("spark.shuffle.push.based.enabled")
```

Comment:
seems like push.based doesn't need to be separate, If all the rest are shuffle.push, then maybe this should just be shuffle.push.enabled.  Or keep them all consistent and call pushBased.enabled, pushBased.ratinedMergeLocations.  Personally I'd say just go with shuffle.push.X